spring-cloud-starter-stream-collector-metrics/src/main/java/org/springframework/cloud/stream/app/metrics/collector/MetricsAggregator.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.metrics.collector;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import com.github.benmanes.caffeine.cache.Cache;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.app.metrics.collector.model.Application;
+import org.springframework.cloud.stream.app.metrics.collector.model.ApplicationMetrics;
+import org.springframework.cloud.stream.app.metrics.collector.model.Instance;
+import org.springframework.cloud.stream.app.metrics.collector.model.Stream;
+import org.springframework.cloud.stream.app.metrics.collector.utils.YANUtils;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+/**
+ * Listens for {@link ApplicationMetrics} coming from the metrics channel. Transform the payload into an {@link Stream} hierarchical model to be consumed by the server.
+ * This class handles conversion, update of deltas for the sendCount, as well as invalidation of stale entries
+ * @author Vinicius Carvalho
+ */
+@Component
+public class MetricsAggregator {
+
+	public static final String OUTPUT_METRIC_NAME = "integration.channel.output.sendRate.mean";
```

Comment:
I think we should at least have a plan for handling non-default channel names so that we know our initial solution will be forward-compatible. Considering this features lives within `org.springframework.cloud.stream` (not dataflow), it seems like we should not be thinking much about "out-of-the-box" apps. Also, I would expect the value of metrics will be even greater for developers creating custom stream apps.